src/Razor/test/RazorLanguage.Test/Legacy/CSharpImplicitExpressionTest.cs

```diff
@@ -217,6 +217,181 @@ public void ParsesNullConditionalOperatorImplicitExpression_Dot14()
             ParseDocumentTest("@val?.more(null ?? true)?.abc");
         }
 
+        [Fact]
+        public void ParsesNullForgivenessOperatorImplicitExpression_DoubleBang_IncompleteBracket()
+        {
+            // Act & Assert
+            ParseDocumentTest("@val!![");
```

Comment:
Can you add some tests to test it within a code block as well as in markup within a functions block?

Reply:
And also inside an attribute value.

Reply:
Sure